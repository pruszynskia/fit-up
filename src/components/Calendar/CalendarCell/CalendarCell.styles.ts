import {makeStyles} from '@material-ui/core/styles'

export default makeStyles(theme => ({
    root: {
        display: "flex",
        flexDirection: "column",
    },
    container: {
        display: "flex",
    },
    column: {
        flexDirection: "column",
        justifyContent: "center",
    },
    row: {
        flexDirection: "row",
        justifyContent: "center",
    },
    offset: {
        marginLeft: "20px"
    },
    hasData: {
        display: "100%",
        width: "100%",
        height: "auto",
        backgroundColor: "#424242 !important",
        color: "#fff",

    },

    selected: {
        borderRadius: "50%",
        background: theme.palette.secondary.main,
        color: "#fff"
    },
    today: {
        borderRadius: "50%",
        background: theme.palette.primary.main,
        color: "#fff",
        fontWeight: "bold",
    },
    day: {
        // height: "100%",
        width: "2.5em",
        height: "2.5em",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        borderRadius: "50%",
    
        transition: "background-color 0.3s ease-in-out",
        "&:hover": {
            backgroundColor: theme.palette.primary.main,
            cursor: "pointer",
            color: "#fff"
        }
    },
}))