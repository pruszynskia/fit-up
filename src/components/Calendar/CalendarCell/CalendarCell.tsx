import React, {useState} from 'react'
import {useSelector} from 'react-redux'
import moment from 'moment'

import useStyles from './CalendarCell.styles'
import classnames from 'classnames';

import {
    Dialog,
    DialogContent,
} from '@material-ui/core';

import ListOfWorkoutDayForm from '../../ListOfWorkoutDayForm';
import {RootState, WorkoutDayDetails} from '../../../lib/types'


interface CalendarCellProps {
    date: moment.Moment;
    selectedDay: any;
    handleDayClick: Function
}

export default function CalendarCell({date, selectedDay, handleDayClick}: CalendarCellProps) {
    const classes = useStyles()

    const data = useSelector((state: RootState) => state.workoutDays)
                                 .filter((ex: WorkoutDayDetails) => ex.date === date.format("DDMMYYYY"))

    const [open, setOpen] = useState<boolean>(false)
    const handleOpen = () => setOpen(true)
    const handleClose = () => setOpen(false)

    return <div>
        <span
            onClick={() => handleDayClick(date)}
            className={classnames(classes.day, {
                [classes.selected]:
                date.format("DDMMYYYY") === selectedDay.format("DDMMYYYY"),
                [classes.today]:
                date.format("DDMMYYYY") === moment().format("DDMMYYYY"),
            })}
        >
            <div className={classnames({
                    [classes.hasData]: Boolean(data.length),
                    })} 
                onClick={handleOpen}
            >
                {date.format("DD")}
            </div>

            <div className={classnames(
                    classes.container,
                    classes.column, 
                )}
            >
                <Dialog
                    open={open} 
                    onClose={handleClose} 
                >
                    <DialogContent>
                        <ListOfWorkoutDayForm 
                            date={date} 
                        />
                    </DialogContent>
                </Dialog>
            </div>
        </span>
    </div>
}