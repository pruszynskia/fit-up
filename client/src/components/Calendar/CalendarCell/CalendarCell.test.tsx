import moment from "moment";
import React from "react";
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import CalendarCell from ".";

jest.mock('./CalendarCell', () => 'CalendarCell')
jest.mock('moment', () => {
    return () => jest.requireActual('moment')('2020-01-01T00:00:00.000Z');
});

interface CalendarCell {
    date: moment.Moment;
    selectedDay: any;
    handleDayClick: Function
}

describe("<CalendarCell/> should render as expected", () => {
    it("Should render as expected", () => {

        let date = moment();
        let selectedDay
        const handleDayClick = () => handleDayClick;
        
        const mockStore = configureStore();
        let store

        store = mockStore(selectedDay)
        const tree = renderer.create(
            <Provider store={store}>
                <CalendarCell date={date} selectedDay={selectedDay} handleDayClick={handleDayClick} />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})