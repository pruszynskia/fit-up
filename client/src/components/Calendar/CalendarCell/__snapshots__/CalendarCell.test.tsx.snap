// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarCell/> should render as expected Should render as expected 1`] = `
<CalendarCell
  date={"2020-01-01T00:00:00.000Z"}
  handleDayClick={[Function]}
/>
`;
